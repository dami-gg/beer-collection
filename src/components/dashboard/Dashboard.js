// @flow

import React, {Component} from 'react';
import DashboardOption from './dashboard-option/DashboardOption';

import './dashboard.scss';

import crate from '../../assets/images/crate.png';
import tap from '../../assets/images/tap.png';

class Dashboard extends Component {
  render() {
    return (
        <div className="dashboard">
          <DashboardOption
              title="Add beer to your collection"
              url="/beer/add"
              image={tap}
          />
          <DashboardOption
              title="See your collection"
              url="/collection"
              image={crate}
          />
        </div>
    );
  }
}

export default Dashboard;
