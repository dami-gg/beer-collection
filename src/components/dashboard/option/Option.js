// @flow

import React, {PureComponent} from 'react';
import {Link} from 'react-router-dom';

import './option.scss';

class Option extends PureComponent {
  props: {
    title: string,
    url: string,
    image: string
  };

  render() {
    return (
        <div className="dashboard__option">
          <Link className="dashboard__option__card" to={this.props.url}>
            <div className="dashboard__option__image-wrapper">
              <img src={this.props.image} className="dashboard__option__image" alt="logo"/>
            </div>
            <div className="dashboard__option__title-wrapper">
              <p className="dashboard__option__title">{this.props.title}</p>
            </div>
          </Link>
        </div>
    );
  }
}

export default Option;
