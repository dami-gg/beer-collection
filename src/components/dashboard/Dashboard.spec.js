import React from 'react';
import Dashboard from './Dashboard';
import Option from './option/Option';

import {StaticRouter as Router} from 'react-router-dom';

import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe('Dashboard', () => {
  let component,
      element,
      mock;

  beforeEach(mocks);

  it('should render without crashing', () => {
    shallow(<Dashboard />);
  });

  it('should match the snapshot', () => {
    component = renderer.create(
        <Router context={mock.context}>
          <Dashboard/>
        </Router>
    );
    let json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  describe('render', () => {
    beforeEach(setup);

    it('should render a dashboard element', () => {
      element = component.find('.dashboard');

      expect(element.length).toEqual(1);
    });

    it('should render four Option components', () => {
      element = component.find(Option);

      expect(element.length).toEqual(4);
    });
  });

  function setup() {
    render();
  }

  function render() {
    component = shallow(<Dashboard />);
  }

  function mocks() {
    mock = {
      context: {}
    };
  }
});

