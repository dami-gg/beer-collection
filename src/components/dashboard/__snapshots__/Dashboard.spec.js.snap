// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match the snapshot 1`] = `
<div
  className="dashboard"
>
  <div
    className="dashboard__option"
  >
    <a
      className="dashboard__option__card"
      href="/beer/add"
      onClick={[Function]}
    >
      <div
        className="dashboard__option__image-wrapper"
      >
        <img
          alt="logo"
          className="dashboard__option__image"
          src="test-file-stub"
        />
      </div>
      <div
        className="dashboard__option__title-wrapper"
      >
        <p
          className="dashboard__option__title"
        >
          Add a new beer
        </p>
      </div>
    </a>
  </div>
  <div
    className="dashboard__option"
  >
    <a
      className="dashboard__option__card"
      href="/collection"
      onClick={[Function]}
    >
      <div
        className="dashboard__option__image-wrapper"
      >
        <img
          alt="logo"
          className="dashboard__option__image"
          src="test-file-stub"
        />
      </div>
      <div
        className="dashboard__option__title-wrapper"
      >
        <p
          className="dashboard__option__title"
        >
          See your collection
        </p>
      </div>
    </a>
  </div>
  <div
    className="dashboard__option"
  >
    <a
      className="dashboard__option__card"
      href="/statistics"
      onClick={[Function]}
    >
      <div
        className="dashboard__option__image-wrapper"
      >
        <img
          alt="logo"
          className="dashboard__option__image"
          src="test-file-stub"
        />
      </div>
      <div
        className="dashboard__option__title-wrapper"
      >
        <p
          className="dashboard__option__title"
        >
          See statistics
        </p>
      </div>
    </a>
  </div>
  <div
    className="dashboard__option"
  >
    <a
      className="dashboard__option__card"
      href="/map"
      onClick={[Function]}
    >
      <div
        className="dashboard__option__image-wrapper"
      >
        <img
          alt="logo"
          className="dashboard__option__image"
          src="test-file-stub"
        />
      </div>
      <div
        className="dashboard__option__title-wrapper"
      >
        <p
          className="dashboard__option__title"
        >
          Beer map
        </p>
      </div>
    </a>
  </div>
</div>
`;
